lib/Db/MessageMapper.php

```diff
@@ -954,28 +945,33 @@ public function findRelatedData(array $messages, string $userId): array {
 	 * @return int[]
 	 */
 	public function findNewIds(Mailbox $mailbox, array $ids): array {
-		$sub = $this->db->getQueryBuilder();
-		$qb = $this->db->getQueryBuilder();
+		$select = $this->db->getQueryBuilder();
+		$subSelect = $this->db->getQueryBuilder();
+
+		$knowIdsParameter = $select->createParameter('ids');
 
-		$subSelect = $sub
-			->select($sub->func()->max('uid'))
+		$subSelect
+			->select($subSelect->func()->min('sent_at'))
 			->from($this->getTableName())
 			->where(
-				$sub->expr()->eq('mailbox_id', $qb->createNamedParameter($mailbox->getId(), IQueryBuilder::PARAM_INT)),
-				$sub->expr()->in('id', $qb->createParameter('ids'))
+				$subSelect->expr()->eq('mailbox_id', $select->createNamedParameter($mailbox->getId(), IQueryBuilder::PARAM_INT)),
+				$subSelect->expr()->in('id', $knowIdsParameter)
 			);
 
-		$qb->select('id')
-			->from($this->getTableName())
+		$select
```

Comment:
- [ ] https://github.com/nextcloud/mail/pull/4947 introduced chunking for in expressions. Check the findNewIds query to make sure it still works es expected.